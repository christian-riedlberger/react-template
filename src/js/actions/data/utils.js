// @flow
import faker from 'faker';

/**
 * choose a random item from an array
 */
export const choice = array => array[faker.random.number(array.length - 1)];

/**
 * create an array of n sized, and fill with the results of the function
 */
export const fill = (len: Number, func: Function) =>
    Array(len || 1)
        .fill(null)
        .map(func);

/**
 * get value from localStorage, orreturn default value
 */
export const getFromStorage = (domain: string, defaultValue: any) => {
    const raw = localStorage.getItem(domain);
    const value = JSON.parse(raw);
    if (!value) return defaultValue;
    return value;
};

/**
 * return data after delay elapsed
 */
export const sleep = (delay, obj) =>
    // eslint-disable-next-line compat/compat
    new Promise(resolve =>
        setTimeout(
            () =>
                resolve({
                    data: obj
                }),
            delay
        )
    );
