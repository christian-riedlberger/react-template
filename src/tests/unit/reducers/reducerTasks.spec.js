import reducerTasks, { initialState } from 'reducers/reducerTasks';
import {
    FETCH_ISSUED_TASKS,
    FETCH_RECEIVED_TASKS
    // CLEAR_RECEIVED_TASKS,
    // CLEAR_ISSUED_TASKS
} from 'constants/ActionTypes';

// eslint-disable-next-line import/extensions
import data from './tasks.json';

describe('reducerTasks.js', () => {
    describe(FETCH_ISSUED_TASKS, () => {
        const fetchTasks = responseData => {
            const action = {
                type: `${FETCH_ISSUED_TASKS}_FULFILLED`,
                payload: { data: responseData }
            };

            return reducerTasks(initialState, action);
        };

        it('should handle issued cert tasks', () => {
            expect(fetchTasks(data.certTasksIssued)).toMatchSnapshot();
            expect(
                fetchTasks(data.certTasksIssuedWithoutRefs)
            ).toMatchSnapshot();
        });

        it('should handle recieved spec tasks', () => {
            expect(fetchTasks(data.specTasksIssued)).toMatchSnapshot();
        });
    });

    describe(FETCH_RECEIVED_TASKS, () => {
        const fetchTasks = responseData => {
            const action = {
                type: `${FETCH_RECEIVED_TASKS}_FULFILLED`,
                payload: { data: responseData }
            };

            return reducerTasks(initialState, action);
        };

        it('should handle recieved cert tasks', () => {
            expect(fetchTasks(data.certTasksRecieved)).toMatchSnapshot();
            expect(
                fetchTasks(data.certTasksRecievedWithoutRefs)
            ).toMatchSnapshot();
        });

        it('should handle recieved spec tasks', () => {
            expect(fetchTasks(data.specTasksRecieved)).toMatchSnapshot();
        });
    });
});
