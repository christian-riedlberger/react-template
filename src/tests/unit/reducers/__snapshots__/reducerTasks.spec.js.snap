// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducerTasks.js FETCH_ISSUED_TASKS should handle issued cert tasks 1`] = `
Object {
  "activeTask": null,
  "alltasks": Array [],
  "assignedBy": undefined,
  "assignee": undefined,
  "counts": Object {},
  "definitions": Array [],
  "initialized": false,
  "instance": null,
  "instanceTasks": Array [],
  "isAllTasksLoading": false,
  "isLoading": false,
  "isLoadingPackage": false,
  "isLoadingTaskAuthorities": false,
  "isLoadingWorkflowDetails": false,
  "issuedTasks": Array [
    Object {
      "additionalInfo": null,
      "collaborators": Array [],
      "collection": "Certification",
      "comment": undefined,
      "completionProgress": "0/1",
      "definitionUrl": "api/workflow-definitions/activiti$certificationRequest:19:8516",
      "details": null,
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9162/diagram?alf_ticket=",
      "due": "2020-08-26T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": false,
      "isReassignable": true,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/task/details/activiti$start9162",
      "name": "cert1",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/d6fbe39a-0e54-4612-b44e-e19d55edbead",
      "packageId": "workspace://SpacesStore/d6fbe39a-0e54-4612-b44e-e19d55edbead",
      "priority": "1",
      "processId": "activiti$certificationRequest",
      "progress": "notStarted",
      "stage": "cert:submitCertificationTask",
      "start": "2020-07-06T17:40:43.669Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$start9162",
      "title": "cert:submitCertificationTask",
      "tskName": "cert:submitCertificationTask",
      "wfInstanceId": "activiti$9162",
    },
    Object {
      "additionalInfo": "asdf",
      "collaborators": Array [],
      "collection": "Certification",
      "comment": undefined,
      "completionProgress": "0/1",
      "definitionUrl": "api/workflow-definitions/activiti$certificationRequest:19:8516",
      "details": "test description",
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9248/diagram?alf_ticket=",
      "due": "2020-08-26T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": false,
      "isReassignable": true,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/task/details/activiti$start9248",
      "name": "cert 2",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/6b635988-eccc-45ad-8107-ca92e5c81295",
      "packageId": "workspace://SpacesStore/6b635988-eccc-45ad-8107-ca92e5c81295",
      "priority": "1",
      "processId": "activiti$certificationRequest",
      "progress": "notStarted",
      "stage": "cert:submitCertificationTask",
      "start": "2020-07-06T17:42:22.780Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$start9248",
      "title": "cert:submitCertificationTask",
      "tskName": "cert:submitCertificationTask",
      "wfInstanceId": "activiti$9248",
    },
  ],
  "issuedTasksIsLoading": false,
  "issuedTotalTasks": undefined,
  "needsTaskReset": false,
  "organizations": undefined,
  "organizationsBy": Array [],
  "organizationsTo": Array [],
  "packages": Array [],
  "receivedTasks": Array [],
  "receivedTasksIsLoading": false,
  "receivedTotalTasks": 0,
  "recentTasks": null,
  "tasks": Array [],
  "totalTasks": 0,
}
`;

exports[`reducerTasks.js FETCH_ISSUED_TASKS should handle issued cert tasks 2`] = `
Object {
  "activeTask": null,
  "alltasks": Array [],
  "assignedBy": undefined,
  "assignee": undefined,
  "counts": Object {},
  "definitions": Array [],
  "initialized": false,
  "instance": null,
  "instanceTasks": Array [],
  "isAllTasksLoading": false,
  "isLoading": false,
  "isLoadingPackage": false,
  "isLoadingTaskAuthorities": false,
  "isLoadingWorkflowDetails": false,
  "issuedTasks": Array [
    Object {
      "additionalInfo": null,
      "collaborators": Array [],
      "collection": "Certification",
      "comment": undefined,
      "completionProgress": "0/1",
      "definitionUrl": "api/workflow-definitions/activiti$certificationRequest:19:8516",
      "details": null,
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9162/diagram?alf_ticket=",
      "due": "2020-08-26T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": false,
      "isReassignable": true,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/task/details/activiti$start9162",
      "name": "cert1",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/d6fbe39a-0e54-4612-b44e-e19d55edbead",
      "packageId": "workspace://SpacesStore/d6fbe39a-0e54-4612-b44e-e19d55edbead",
      "priority": "1",
      "processId": "activiti$certificationRequest",
      "progress": "notStarted",
      "stage": "cert:submitCertificationTask",
      "start": "2020-07-06T17:40:43.669Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$start9162",
      "title": "cert:submitCertificationTask",
      "tskName": "cert:submitCertificationTask",
      "wfInstanceId": "activiti$9162",
    },
    Object {
      "additionalInfo": "asdf",
      "collaborators": Array [],
      "collection": "Certification",
      "comment": undefined,
      "completionProgress": "0/1",
      "definitionUrl": "api/workflow-definitions/activiti$certificationRequest:19:8516",
      "details": "test description",
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9248/diagram?alf_ticket=",
      "due": "2020-08-26T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": false,
      "isReassignable": true,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/task/details/activiti$start9248",
      "name": "cert 2",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/6b635988-eccc-45ad-8107-ca92e5c81295",
      "packageId": "workspace://SpacesStore/6b635988-eccc-45ad-8107-ca92e5c81295",
      "priority": "1",
      "processId": "activiti$certificationRequest",
      "progress": "notStarted",
      "stage": "cert:submitCertificationTask",
      "start": "2020-07-06T17:42:22.780Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$start9248",
      "title": "cert:submitCertificationTask",
      "tskName": "cert:submitCertificationTask",
      "wfInstanceId": "activiti$9248",
    },
  ],
  "issuedTasksIsLoading": false,
  "issuedTotalTasks": undefined,
  "needsTaskReset": false,
  "organizations": undefined,
  "organizationsBy": Array [],
  "organizationsTo": Array [],
  "packages": Array [],
  "receivedTasks": Array [],
  "receivedTasksIsLoading": false,
  "receivedTotalTasks": 0,
  "recentTasks": null,
  "tasks": Array [],
  "totalTasks": 0,
}
`;

exports[`reducerTasks.js FETCH_ISSUED_TASKS should handle recieved spec tasks 1`] = `
Object {
  "activeTask": null,
  "alltasks": Array [],
  "assignedBy": undefined,
  "assignee": undefined,
  "counts": Object {},
  "definitions": Array [],
  "initialized": false,
  "instance": null,
  "instanceTasks": Array [],
  "isAllTasksLoading": false,
  "isLoading": false,
  "isLoadingPackage": false,
  "isLoadingTaskAuthorities": false,
  "isLoadingWorkflowDetails": false,
  "issuedTasks": Array [
    Object {
      "additionalInfo": "info",
      "collaborators": Array [],
      "collection": "Compliant",
      "comment": undefined,
      "completionProgress": "0/4",
      "definitionUrl": "api/workflow-definitions/activiti$compliantWorkflow:19:8508",
      "details": "sadf",
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9635/diagram?alf_ticket=",
      "due": "2020-08-27T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": false,
      "isReassignable": true,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/workflow/details/task$start9635,task$start9734,task$start9833,task$start9932",
      "name": "spec 2",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/d2743f33-04c3-4554-b602-cd671f8b6ca5",
      "packageId": "workspace://SpacesStore/d2743f33-04c3-4554-b602-cd671f8b6ca5",
      "packageRefs": Array [
        "workspace://SpacesStore/4aeaa213-a3c8-46c9-89c1-dd03befc697e",
        "workspace://SpacesStore/d165f762-9ea6-48a0-b5f5-bba4833b5625",
        "workspace://SpacesStore/06a5c1b0-1047-4abf-843b-b160ac037e53",
        "workspace://SpacesStore/89cc7f09-4b93-44ac-9930-2bad785fd13c",
      ],
      "priority": "1",
      "processId": "activiti$compliantWorkflow",
      "progress": "notStarted",
      "stage": "gfr:submitReviewTask",
      "start": "2020-07-06T20:38:12.337Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$start9635",
      "taskRefs": Array [
        "activiti$start9635",
        "activiti$start9734",
        "activiti$start9833",
        "activiti$start9932",
      ],
      "title": "gfr:submitReviewTask",
      "tskName": "gfr:submitReviewTask",
      "wfInstanceId": "activiti$9635",
    },
    Object {
      "additionalInfo": "ASDF",
      "collaborators": Array [],
      "collection": "Compliant",
      "comment": undefined,
      "completionProgress": "0/3",
      "definitionUrl": "api/workflow-definitions/activiti$compliantWorkflow:19:8508",
      "details": "test",
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9338/diagram?alf_ticket=",
      "due": "2020-08-06T05:59:00.450Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": false,
      "isReassignable": true,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/workflow/details/task$start9338,task$start9437,task$start9536",
      "name": "spec 1",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/186c90ed-59d7-4c29-a5d8-7ef216ae944d",
      "packageId": "workspace://SpacesStore/186c90ed-59d7-4c29-a5d8-7ef216ae944d",
      "packageRefs": Array [
        "workspace://SpacesStore/55787a97-8390-4395-8921-f4c1e25bb2ab",
        "workspace://SpacesStore/13ce63c7-ab5b-4c05-bf85-dda86f1a1b26",
        "workspace://SpacesStore/21ad80c8-239e-42e0-b897-4a6188df43e2",
      ],
      "priority": "1",
      "processId": "activiti$compliantWorkflow",
      "progress": "inprogress",
      "stage": "gfr:submitReviewTask",
      "start": "2020-07-06T18:33:31.849Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$start9338",
      "taskRefs": Array [
        "activiti$start9338",
        "activiti$start9437",
        "activiti$start9536",
      ],
      "title": "gfr:submitReviewTask",
      "tskName": "gfr:submitReviewTask",
      "wfInstanceId": "activiti$9338",
    },
  ],
  "issuedTasksIsLoading": false,
  "issuedTotalTasks": undefined,
  "needsTaskReset": false,
  "organizations": undefined,
  "organizationsBy": Array [],
  "organizationsTo": Array [],
  "packages": Array [],
  "receivedTasks": Array [],
  "receivedTasksIsLoading": false,
  "receivedTotalTasks": 0,
  "recentTasks": null,
  "tasks": Array [],
  "totalTasks": 0,
}
`;

exports[`reducerTasks.js FETCH_RECEIVED_TASKS should handle recieved cert tasks 1`] = `
Object {
  "activeTask": null,
  "alltasks": Array [],
  "assignedBy": undefined,
  "assignee": undefined,
  "counts": Object {},
  "definitions": Array [],
  "initialized": false,
  "instance": null,
  "instanceTasks": Array [],
  "isAllTasksLoading": false,
  "isLoading": false,
  "isLoadingPackage": false,
  "isLoadingTaskAuthorities": false,
  "isLoadingWorkflowDetails": false,
  "issuedTasks": Array [],
  "issuedTasksIsLoading": false,
  "issuedTotalTasks": 0,
  "needsTaskReset": false,
  "organizations": undefined,
  "organizationsBy": Array [],
  "organizationsTo": Array [],
  "packages": Array [],
  "receivedTasks": Array [
    Object {
      "additionalInfo": "asdf",
      "collaborators": Array [],
      "collection": "Certification",
      "comment": undefined,
      "completionProgress": "0/1",
      "definitionUrl": "api/workflow-definitions/activiti$certificationRequest:19:8516",
      "details": "test description",
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9248/diagram?alf_ticket=",
      "due": "2020-08-26T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": true,
      "isReassignable": null,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/task/details/activiti$9308",
      "name": "cert 2",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/6b635988-eccc-45ad-8107-ca92e5c81295",
      "packageId": "workspace://SpacesStore/6b635988-eccc-45ad-8107-ca92e5c81295",
      "priority": "1",
      "processId": "activiti$certificationRequest",
      "progress": "notStarted",
      "stage": "Supplier to provide certificate",
      "start": "2020-07-06T17:42:22.780Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$9308",
      "title": "Supplier to provide certificate",
      "tskName": "cert:supplierCertificationIntake",
      "wfInstanceId": "activiti$9248",
    },
    Object {
      "additionalInfo": null,
      "collaborators": Array [],
      "collection": "Certification",
      "comment": undefined,
      "completionProgress": "0/1",
      "definitionUrl": "api/workflow-definitions/activiti$certificationRequest:19:8516",
      "details": null,
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9162/diagram?alf_ticket=",
      "due": "2020-08-26T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": true,
      "isReassignable": null,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/task/details/activiti$9218",
      "name": "cert1",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/d6fbe39a-0e54-4612-b44e-e19d55edbead",
      "packageId": "workspace://SpacesStore/d6fbe39a-0e54-4612-b44e-e19d55edbead",
      "priority": "1",
      "processId": "activiti$certificationRequest",
      "progress": "notStarted",
      "stage": "Supplier to provide certificate",
      "start": "2020-07-06T17:40:43.669Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$9218",
      "title": "Supplier to provide certificate",
      "tskName": "cert:supplierCertificationIntake",
      "wfInstanceId": "activiti$9162",
    },
  ],
  "receivedTasksIsLoading": false,
  "receivedTotalTasks": undefined,
  "recentTasks": null,
  "tasks": Array [],
  "totalTasks": 0,
}
`;

exports[`reducerTasks.js FETCH_RECEIVED_TASKS should handle recieved cert tasks 2`] = `
Object {
  "activeTask": null,
  "alltasks": Array [],
  "assignedBy": undefined,
  "assignee": undefined,
  "counts": Object {},
  "definitions": Array [],
  "initialized": false,
  "instance": null,
  "instanceTasks": Array [],
  "isAllTasksLoading": false,
  "isLoading": false,
  "isLoadingPackage": false,
  "isLoadingTaskAuthorities": false,
  "isLoadingWorkflowDetails": false,
  "issuedTasks": Array [],
  "issuedTasksIsLoading": false,
  "issuedTotalTasks": 0,
  "needsTaskReset": false,
  "organizations": undefined,
  "organizationsBy": Array [],
  "organizationsTo": Array [],
  "packages": Array [],
  "receivedTasks": Array [
    Object {
      "additionalInfo": "asdf",
      "collaborators": Array [],
      "collection": "Certification",
      "comment": undefined,
      "completionProgress": "0/1",
      "definitionUrl": "api/workflow-definitions/activiti$certificationRequest:19:8516",
      "details": "test description",
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9248/diagram?alf_ticket=",
      "due": "2020-08-26T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": true,
      "isReassignable": null,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/task/details/activiti$9308",
      "name": "cert 2",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/6b635988-eccc-45ad-8107-ca92e5c81295",
      "packageId": "workspace://SpacesStore/6b635988-eccc-45ad-8107-ca92e5c81295",
      "priority": "1",
      "processId": "activiti$certificationRequest",
      "progress": "notStarted",
      "stage": "Supplier to provide certificate",
      "start": "2020-07-06T17:42:22.780Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$9308",
      "title": "Supplier to provide certificate",
      "tskName": "cert:supplierCertificationIntake",
      "wfInstanceId": "activiti$9248",
    },
    Object {
      "additionalInfo": null,
      "collaborators": Array [],
      "collection": "Certification",
      "comment": undefined,
      "completionProgress": "0/1",
      "definitionUrl": "api/workflow-definitions/activiti$certificationRequest:19:8516",
      "details": null,
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9162/diagram?alf_ticket=",
      "due": "2020-08-26T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": true,
      "isReassignable": null,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/task/details/activiti$9218",
      "name": "cert1",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/d6fbe39a-0e54-4612-b44e-e19d55edbead",
      "packageId": "workspace://SpacesStore/d6fbe39a-0e54-4612-b44e-e19d55edbead",
      "priority": "1",
      "processId": "activiti$certificationRequest",
      "progress": "notStarted",
      "stage": "Supplier to provide certificate",
      "start": "2020-07-06T17:40:43.669Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$9218",
      "title": "Supplier to provide certificate",
      "tskName": "cert:supplierCertificationIntake",
      "wfInstanceId": "activiti$9162",
    },
  ],
  "receivedTasksIsLoading": false,
  "receivedTotalTasks": undefined,
  "recentTasks": null,
  "tasks": Array [],
  "totalTasks": 0,
}
`;

exports[`reducerTasks.js FETCH_RECEIVED_TASKS should handle recieved spec tasks 1`] = `
Object {
  "activeTask": null,
  "alltasks": Array [],
  "assignedBy": undefined,
  "assignee": undefined,
  "counts": Object {},
  "definitions": Array [],
  "initialized": false,
  "instance": null,
  "instanceTasks": Array [],
  "isAllTasksLoading": false,
  "isLoading": false,
  "isLoadingPackage": false,
  "isLoadingTaskAuthorities": false,
  "isLoadingWorkflowDetails": false,
  "issuedTasks": Array [],
  "issuedTasksIsLoading": false,
  "issuedTotalTasks": 0,
  "needsTaskReset": false,
  "organizations": undefined,
  "organizationsBy": Array [],
  "organizationsTo": Array [],
  "packages": Array [],
  "receivedTasks": Array [
    Object {
      "additionalInfo": "info",
      "collaborators": Array [],
      "collection": "Compliant",
      "comment": undefined,
      "completionProgress": "0/4",
      "definitionUrl": "api/workflow-definitions/activiti$compliantWorkflow:19:8508",
      "details": "sadf",
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9635/diagram?alf_ticket=",
      "due": "2020-08-27T05:59:00.000Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": true,
      "isReassignable": null,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/workflow/details/task$9704,task$9803,task$9902,task$10001",
      "name": "spec 2",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/d2743f33-04c3-4554-b602-cd671f8b6ca5",
      "packageId": "workspace://SpacesStore/d2743f33-04c3-4554-b602-cd671f8b6ca5",
      "packageRefs": Array [
        "workspace://SpacesStore/4aeaa213-a3c8-46c9-89c1-dd03befc697e",
        "workspace://SpacesStore/d165f762-9ea6-48a0-b5f5-bba4833b5625",
        "workspace://SpacesStore/06a5c1b0-1047-4abf-843b-b160ac037e53",
        "workspace://SpacesStore/89cc7f09-4b93-44ac-9930-2bad785fd13c",
      ],
      "priority": "1",
      "processId": "activiti$compliantWorkflow",
      "progress": "notStarted",
      "stage": "Review Request",
      "start": "2020-07-06T20:38:12.337Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$9704",
      "taskRefs": Array [
        "activiti$9704",
        "activiti$9803",
        "activiti$9902",
        "activiti$10001",
      ],
      "title": "Review Request",
      "tskName": "gfr:stdReview",
      "wfInstanceId": "activiti$9635",
    },
    Object {
      "additionalInfo": "ASDF",
      "collaborators": Array [],
      "collection": "Compliant",
      "comment": undefined,
      "completionProgress": "0/3",
      "definitionUrl": "api/workflow-definitions/activiti$compliantWorkflow:19:8508",
      "details": "test",
      "diagramURL": "ERROR-DOMAIN/alfresco/s/api/workflow-instances/activiti$9338/diagram?alf_ticket=",
      "due": "2020-08-06T05:59:00.450Z",
      "entity": "RECEIVERCYPRESSTEST",
      "errors": Array [],
      "initiator": "Cy Creator",
      "initiatorFirstName": "Cy",
      "initiatorLastName": "Creator",
      "initiatorUsername": "test_user_cypress_creator",
      "isPooled": true,
      "isReassignable": null,
      "issuingEntity": "CREATORCYPRESSTEST",
      "link": "/workflow/details/task$9407,task$9506,task$9605",
      "name": "spec 1",
      "owner": "RECEIVERCYPRESSTEST",
      "package": "workspace://SpacesStore/186c90ed-59d7-4c29-a5d8-7ef216ae944d",
      "packageId": "workspace://SpacesStore/186c90ed-59d7-4c29-a5d8-7ef216ae944d",
      "packageRefs": Array [
        "workspace://SpacesStore/55787a97-8390-4395-8921-f4c1e25bb2ab",
        "workspace://SpacesStore/13ce63c7-ab5b-4c05-bf85-dda86f1a1b26",
        "workspace://SpacesStore/21ad80c8-239e-42e0-b897-4a6188df43e2",
      ],
      "priority": "1",
      "processId": "activiti$compliantWorkflow",
      "progress": "inprogress",
      "stage": "Review Request",
      "start": "2020-07-06T18:33:31.849Z",
      "state": "IN_PROGRESS",
      "status": "pending",
      "taskId": "activiti$9407",
      "taskRefs": Array [
        "activiti$9407",
        "activiti$9506",
        "activiti$9605",
      ],
      "title": "Review Request",
      "tskName": "gfr:stdReview",
      "wfInstanceId": "activiti$9338",
    },
  ],
  "receivedTasksIsLoading": false,
  "receivedTotalTasks": undefined,
  "recentTasks": null,
  "tasks": Array [],
  "totalTasks": 0,
}
`;
